### PR TITLE
Red 365 v2 registration

### PR DESCRIPTION
**Pull request by Ivan-dev-RED**

### CODE REVIEW

File: src/services/resolvers/ens-resolver.service.ts [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+import { ResolverService } from "@resolver/services/resolvers/resolver.service";
 import EvmWeb3Service from "@resolver/services/web3/evm-web3.service";
 import type { Network, Account } from "@resolver/models/types";
 
 
-export class EnsResolverService implements ResolverService {
+export class EnsResolverService extends ResolverService {
 
-    supportedNetworks: Network[] = ["eth", "bsc"];
+    getSupportedNetworks(): Network[] {
```
Comment (au-e2xlabs) [inline src/services/resolvers/ens-resolver.service.ts:8]: i think we should create this abstract method in ResolverService abstract class

File: src/services/resolvers/resolver.service.ts [hunk 2]
```diff
@@ -1,5 +1,14 @@
-import type { Network, Account } from "@resolver/models/types";
+import type { Nodes, Account,Network } from "@resolver/models/types";
+import { flatten } from "lodash";
 
-export interface ResolverService {
-    resolve(domain: string, network: Network, nodeLink: string): Promise<Account[]>;
+export abstract class ResolverService {
+
+    abstract getSupportedNetworks(): Network[]
+
```
Comment (au-e2xlabs) [inline src/services/resolvers/resolver.service.ts:7]:
```
isSupportedNetwork(network: Network) {
        if (!this.getSupportedNetworks().some(it => it === network)) {
            console.log(`${network} not supported by ${this.constructor.toString().match(/\w+/g)![1].replace("ResolverService", "")}.`);
            return [];
        }
    }
```
and use it in all services

File: src/services/resolvers/resolver.service.ts [hunk 3]
```diff
@@ -1,5 +1,14 @@
-import type { Network, Account } from "@resolver/models/types";
+import type { Nodes, Account,Network } from "@resolver/models/types";
+import { flatten } from "lodash";
 
-export interface ResolverService {
-    resolve(domain: string, network: Network, nodeLink: string): Promise<Account[]>;
+export abstract class ResolverService {
+
+    abstract getSupportedNetworks(): Network[]
```
[the remark this answers is not in the document]
Comment (Ivan-dev-RED) [reply src/services/resolvers/resolver.service.ts:6]: ok

File: src/services/resolvers/resolver.service.ts (hunk 2 above)
Comment (Ivan-dev-RED) [reply src/services/resolvers/resolver.service.ts:7]: ok, but no

**Review by au-e2xlabs — APPROVED** [review]

File: src/services/resolvers/resolver.service.ts (hunk 2 above)
Comment (au-e2xlabs) [reply src/services/resolvers/resolver.service.ts:7]:
```
isSupportedNetwork(network: Network) {
        if (!this.getSupportedNetworks().some(it => it === network)) {
            console.log(`${network} not supported by ${this.constructor.toString().match(/\w+/g)![1].replace("ResolverService", "")}.`);
            return false;
        }
return true;
    }
```

**Pull request merged**